src/Tizen.NUI/src/public/Window/WindowEvent.cs

```diff
@@ -1082,5 +1083,108 @@ public void VisibiltyChangedSignalEmit(bool visibility)
             }
             VisibilityChangedEventSignal.Emit(this, visibility);
         }
+
+        /// <summary>
+        /// Auxiliary Message Event arguments
+        /// This has the three members as key, value and options.
+        /// Options has the list of string
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public class AuxiliaryMessageEventArgs : EventArgs
+        {
+            private string key;
+            private string val;
+            private List<string> options;
+
+            public string Key
+            {
+                get => key;
+                internal set => key = value;
+            }
+
+            public string Value
+            {
+                get => val;
+                internal set => val = value;
+            }
+
+            public List<string> Options
+            {
+                get => options;
+                internal set => options = value;
+            }
+        }
+
+        private void OnAuxiliaryMessage(string key, string value, IntPtr optionsArray)
+        {
+            // at this case, cMemoryOwn is false, because this is just a test implementation, 
+            // so I create local Dali::Property::Array in dali adaptor's window.
+            // therefore, the PropertyArray needs be managed by Dali, and in nui, no need to free it.
+            using var tmp = new PropertyArray(optionsArray, false);
+            var size = tmp.Size();
+
+            List<string> tmpList = new List<string>();
+
+            for (int i = 0; i < size; i++)
+            {
+                string option = "none";
+                tmp.GetElementAt((uint)i).Get(out option);
+                tmpList.Add(option);
+            }
+
```

Comment:
please call `tmp.Dispose();` here to remove this warning =>

**src/Tizen.NUI/src/public/Window/WindowEvent.cs#L1131
Call System.IDisposable.Dispose on object created by 'tmp.GetElementAt((uint)i)' before all references to it are out of scope**

Reply:
Thank you for your detail review.
I updated that.